src/app/resources/resources-view.component.ts

```diff
@@ -46,12 +48,18 @@ export class ResourcesViewComponent implements OnInit {
   getResource(id: string) {
     return this.couchService.get('resources/' + id)
       .then((data) => {
+        // openWhichFile is used to label which file to start with for HTML resources
+        const filename = data.openWhichFile || Object.keys(data._attachments)[0];
         this.mediaType = data.mediaType;
-        this.contentType = data._attachments[id].content_type;
-        this.resourceSrc = this.urlPrefix + data._id + '/' + data.filename;
+        this.contentType = data._attachments[filename].content_type;
+        this.resourceSrc = this.urlPrefix + data._id + '/' + filename;
         if (this.mediaType === 'pdf') {
           this.pdfSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.resourceSrc);
         }
+        if (this.mediaType === 'zip') {

```

Comment:
it should just use resourceSrc which is the same path. Instead make frame within html view.